src/provider/AppDirectory.ts

```diff
@@ -42,6 +42,10 @@ export class AppDirectory {
     }
 
     private async request(): Promise<Application[]> {
+        // If using the demo app directory in production, return an empty array.
+        if (process.env.NODE_ENV === 'production' && AppDirectory.URL === 'http://localhost:3923/provider/app-directory.json') {

```

Comment:
Does this work? This file runs in openfin, so we won't have `process.env` right? Unless web pack does something clever with it that I'm not aware of...

Reply:
It works on my machine? When you specify a `mode` for webpack (`production` or `development`), it drops that into `process.env.NODE_ENV`, so you can check against it. I think it also will pull those lines out of your built code if you use those variables in if blocks like this, but I'd need to get more into the specifics on that. Can someone test on their machine if they have access to `process.env.NODE_ENV`?

Reply:
heh, yeah - seems webpack does some string replacement here.

This is going to be a temporary mechanism, so going to file this one under "it's weird, but it works".

Reply:
Fair enough. If it works then fine by me. This seems like it could be useful in other places, but a discussion for another time. Webpack never ceases to amaze me with it's magic!